WcaOnRails/spec/requests/users_spec.rb

```diff
@@ -23,6 +23,22 @@
     expect(response.body).to include(I18n.t('devise.confirmations.send_instructions'))
   end
 
+  it 'cannot change password when not recently authenticated' do
+    user = FactoryBot.create :user
+
+    # sign in
+    post user_session_path, params: { 'user[login]' => user.email, 'user[password]' => user.password }
+    follow_redirect!
+    expect(response).to be_successful
+    get profile_edit_path
+    expect(response).to be_successful
+
+    new_password = 'new_password'
+    put user_path(user), params: { 'user[email]' => user.email, 'user[password]' => new_password, 'user[password_confirmation]' => new_password }
+    follow_redirect!
+    expect(response.body).to include(I18n.t('users.edit.sensitive.identity_error'))
+  end
```

Comment:
This is actually kinda weird as we try editing password right after signing in, so intuitively we are recently authenticated. But if that's expected I'm fine with it ^^

Reply:
Yes I wasn't sure how to handle the case of a user who just logged in.
It makes sense to mark them as recently authenticated so I've hooked a block setting the recently_authenticated date upon login.